libexec/pyenv-which

```diff
@@ -86,15 +86,17 @@ else
 
   echo "pyenv: $PYENV_COMMAND: command not found" >&2
 
-  versions="$(pyenv-whence "$PYENV_COMMAND" || true)"
-  if [ -n "$versions" ]; then
-    { echo
-      echo "The \`$1' command exists in these Python versions:"
-      echo "$versions" | sed 's/^/  /g'
-      echo
-      echo "Note: See 'pyenv help global' for tips on allowing both"
-      echo "      python2 and python3 to be found."
-    } >&2
+  if [ -z "$_PYENV_SKIP_STDERR" ]; then
```

Comment:
It rather makes sense to make this a meaningful option to the command (e.g. "skip advice" or even "non-verbose").
It can probably be the default if the output is not to a terminal.

Reply:
I like `--skip-advice`, but I think it makes sense to add it to the internal invocations to make things explicit. I'm also not sure how to check in bash if you are outputing to a terminal. In any case, I think the default would only make sense if the output wasn't going anywhere (i.e. `/dev/null`). If the user is trying to parse stderr they should be able to do so without encountering mysterious disappearing outputs because they weren't executing the command in a terminal.

Should the current argument handling:

```bash
# Provide pyenv completions
if [ "$1" = "--complete" ]; then
  exec pyenv-shims --short
fi

if [ "$2" = "--nosystem" ]; then
  system=""
else
  system="system"
fi
```

Be replaced with something like this:

```bash
system="system"
no_advice=""

for args in "$@"
do
  if [ "$args" = "--complete" ]; then
    exec pyenv-shims --short
  fi

  if [ "$args" = "--nosystem" ]; then
    system=""
    shift
  fi

  if [ "$args" = "--no_advice" ]; then
    no_advice="1"
    shift
  fi
done
```

I'm not sure if I should leave the `exec` part as it was. The behaviors are very close, but not exactly the same. The second variant lets the `--complete` argument be in any positions, not just the first.

Reply:
> I'm also not sure how to check in bash if you are outputing to a terminal.

`[[ -t <fd> ]]`

> I like `--skip-advice`, but I think it makes sense to add it to the internal invocations to make things explicit.  In any case, I think the default would only make sense if the output wasn't going anywhere (i.e. `/dev/null`). If the user is trying to parse stderr they should be able to do so without encountering mysterious disappearing outputs because they weren't executing the command in a terminal.

The advice output is not designed to be parsed.
We would however need to add _two_ options in this case -- "skip it" and "show it" (which would be the default) -- so that the user could explicitly choose either.

I agree though that this a further, optional improvement.

Reply:
I pushed up a bunch more commits, and this is ready for a re-review.